model/map/src/main/java/org/keycloak/models/map/common/AbstractMapProviderFactory.java

```diff
@@ -108,13 +106,34 @@ public String getId() {
     }
 
     protected MapStorage<V, M> getStorage(KeycloakSession session) {
-        ProviderFactory<MapStorageProvider> storageProviderFactory = getComponentFactory(session.getKeycloakSessionFactory(),
-          MapStorageProvider.class, storageConfigScope, MapStorageSpi.NAME);
+        ProviderFactory<MapStorageProvider> storageProviderFactory = getProviderFactoryOrComponentFactory(session, storageConfigScope);
         final MapStorageProvider factory = storageProviderFactory.create(session);
-
         return factory.getStorage(modelType);
     }
 
+    public static ProviderFactory<MapStorageProvider> getProviderFactoryOrComponentFactory(KeycloakSession session, Scope storageConfigScope) {
+        ProviderFactory<MapStorageProvider> storageProviderFactory;
+        if (hasRealmSpecificStorage(session, storageConfigScope)) {
```

Comment:
Should this be 
```suggestion
        if (! hasRealmSpecificStorage(session, storageConfigScope)) {
```
?

Reply:
Your're right, I was too fast here. Will try again.

Reply:
Wanted to push the build and see the build advance while I'm in a meeting.